File: Composition/CS/Composition/Algorithm/Inheritance/AveragingAggregator.cs
using System.Collections.Generic;
using System.Linq;

namespace Algorithm.Inheritance
{
    public class AveragingAggregator : PointsAggregator
    {
        public AveragingAggregator(IEnumerable<Measurement> measurements)
            : base(measurements)
        {
        }

        protected override IEnumerable<Measurement> FilterMeasurements(IEnumerable<Measurement> measurements)
        {
            return measurements;
        }

        protected override Measurement AggregateMeasurements(IEnumerable<Measurement> measurements)
        {           
            return new Measurement
                       {
                           X = (int)measurements.Average(m => m.X), 
                           Y = (int)measurements.Average(m => m.Y)
                       };
        }
    }
}

File: Composition/Java/Composition/src/main/java/algorithm/inhereitance/AveragingAggregator.java
package algorithm.inhereitance;

import static algorithm.MeasurementUtil.averageX;
import static algorithm.MeasurementUtil.averageY;

import java.util.Collection;

import algorithm.Measurement;

public class AveragingAggregator extends PointsAggregator {

	public AveragingAggregator(Collection<Measurement> measurements) {
		super(measurements);
	}

	@Override
	protected Collection<Measurement> filterMeasurements() {
		return measurements;
	}

	@Override
	protected Measurement aggregateMeasurements() {
		return new Measurement(averageX(measurements), averageY(measurements));

	}
}
